lib/spack/spack/package.py

```diff
@@ -2607,27 +2647,32 @@ def _run_default_install_time_test_callbacks(self):
         if self.install_time_test_callbacks is None:
             return
 
-        for name in self.install_time_test_callbacks:
-            try:
-                fn = getattr(self, name)
-            except AttributeError:
-                msg = 'RUN-TESTS: method not implemented [{0}]'
-                tty.warn(msg.format(name))
-            else:
-                tty.msg('RUN-TESTS: install-time tests [{0}]'.format(name))
-                fn()
+        # Log running install-time tests to the build log
+        tty.msg('Running install-time tests')
 
+        # Ensure capture of output for tests that use run_test
+        with self._setup_test():
+            for name in self.install_time_test_callbacks:
+                try:
+                    fn = getattr(self, name)
+                except AttributeError:
+                    msg = 'RUN-TESTS: method not implemented [{0}]'
+                    tty.warn(msg.format(name))
+                else:
+                    tty.msg('RUN-TESTS: install-time tests [{0}]'
+                            .format(name))
+                    try:
+                        fn()
+                    except BaseException as exc:
+                        self.test_failures.append((exc, str(exc)))
```

Comment:
Shouldn't the underlying method (`fn`) call `pkg.run_test` which handles this exception catching? Am I missing something?

Reply:
This is the loop for post-install build-time tests, i.e., running the methods explicitly listed in `install_time_test_callbacks`. So they are running package methods.  The `run_test` method takes the name or path of an executable